### PR TITLE
fix: Update ember models table from v1 to v2

### LINKED ISSUE #1092: Update ember-models-table to 2.x
Ref: https://onechiporenko.github.io/ember-models-table/v.2/#/examples

### PR DESCRIPTION
**Pull request by mayank8318**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Update ember-models-table from v1 to v2

#### Changes proposed in this pull request:
- Update dependency version of ember-models-table

Fixes #1092

### CODE REVIEW

**Comment by mayank8318** [comment]
@niranjan94 @geekyd @CosmicCoder96 I think this test is failing for everyone :
screenshot from 2018-05-05 10-01-15

File: package.json [hunk 1]
```diff
@@ -77,7 +77,7 @@
     "ember-math-helpers": "^2.4.1",
     "ember-maybe-import-regenerator": "^0.1.6",
     "ember-metrics": "^0.12.1",
-    "ember-models-table": "^1.13.0",
+    "ember-models-table": "2",
```
Comment (niranjan94) [inline package.json]: Specify a proper range `^2.5.0`

**Comment by mayank8318** [comment]
@niranjan94 I think I missed something because I noticed that a certain test related to `ui-table-select` is failing saying that instead of Ember.Component, Ember.Object is received. I will go through the docs again to see how to do the migration properly and update the PR accordingly.

**Comment by niranjan94** [comment]
@mayank8318 yes please. Always **Test** it locally before making a PR. By test I do not mean just running the test suite, but actually run the application and try out a few basic tasks. 

Just going to `/events/live` would have shown you that the tables fail to render. Please keep the above in mind :)

**Comment by mayank8318** [comment]
Yep I will definitely keep that in mind. Sorry :)

**Comment by mayank8318** [comment]
@niranjan94 @geekyd The table is being rendered but not correctly. From what I could find out, it doesn't have the appropriate class tags :
(Left is the old v1 implementation     Right is the new v2 implementation)
screenshot from 2018-05-07 14-43-42

As is clear some class tags like `ui ui-table` etc are missing. I think this is happening because the v2 version doesn't have `innerTableWrapper` and `outerTableWrapper` which we are overriding for the Semactic UI theme. As a result the table looks like this - 
screenshot from 2018-05-07 15-30-58

I couldn't find any resources for this problem. Please help me here :)

**Comment by mayank8318** [comment]
I had also asked this question in the ember-models-table repository and it has been answered with possible solutions. https://github.com/onechiporenko/ember-models-table/issues/313
@niranjan94 @geekyd I think the second solution is a good enough workaround in our case and we can find a concrete solution later on. Please let me know what you think of this

**Comment by mayank8318** [comment]
The problem seems to arise as the `classes` object is empty here. So the CSS classes are not applied. 
screenshot from 2018-05-08 12-27-35

If I replace the use of `classes` with direct CSS in the `ui-table.hbs` it works but it means we need to hardcode it. Should I go ahead with it?

**Comment by niranjan94** [comment]
@mayank8318 if I remembered right, the v2 version provides semantic UI support. Can't we use that directly instead of attempting to customize ?

**Comment by mayank8318** [comment]
@niranjan94 I have made the necessary changes. Please review :)

**Comment by mayank8318** [comment]
I think I will have to pass themeInstance through all the tables since it is passed in the controller of the table. Should I go ahead with it?

**Comment by niranjan94** [comment]
@mayank8318 dont' pass theme instance in each of the controllers/templates. Just add the themeInstance in the main ui-table component itself.

**Comment by mayank8318** [comment]
@niranjan94 I have made the changes. Please review https://deploy-preview-1093--open-event-frontend.netlify.com/

File: app/components/ui-table.js [hunk 2]
```diff
@@ -1,130 +1,12 @@
...
-
-  rowTemplate: 'components/ui-table/row',
-
-  expandedRowTemplate: 'components/ui-table/expanded-row',
-
-  noDataShowTemplate: 'components/ui-table/no-data',
-
-  allColumnsHiddenTemplate: 'components/ui-table/all-columns-hidden'
-
+  themeInstance: computed(function() {
```
Comment (niranjan94) [inline app/components/ui-table.js]: I don't think this needs to be a computed property. `themeInstance: Semantic.create()` should do.

File: app/controllers/events/list.js [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
     {
       propertyName : 'starts-at',
-      template     : 'components/ui-table/cell/cell-date',
+      component    : 'components/ui-table/cell/cell-date',
       title        : 'Date'
     },
     {
       propertyName     : 'roles',
-      template         : 'components/ui-table/cell/cell-roles',
+      component        : 'components/ui-table/cell/cell-roles',
```
Comment (niranjan94) [inline app/controllers/events/list.js]: Isn't just `ui-table/cell/cell-roles` working ?

File: app/themes/semantic.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import Default from 'ember-models-table/themes/semanticui';
+
+
+export default Default.extend({
+  components: {
+    'pagination-simple'  : 'components/ui-table/simple-pagination',
+    numericPagination    : 'components/ui-table/numeric-pagination',
```
Comment (niranjan94) [inline app/themes/semantic.js]: missing quotes here ?

File: app/themes/semantic.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    'table-footer'       : 'components/ui-table/table-footer',
```
Comment (niranjan94) [inline app/themes/semantic.js]: For all of this, doesn't something like `ui-table/table-footer` work ? (Without the `components/` part)

Comment (mayank8318) [reply app/themes/semantic.js]: I didn't check. Will do it now

Comment (mayank8318) [reply app/themes/semantic.js]: No doesn't work

File: app/controllers/events/list.js (hunk 3 above)
Comment (mayank8318) [reply app/controllers/events/list.js]: Nope

File: app/themes/semantic.js (hunk 5 above)
Comment (niranjan94) [reply app/themes/semantic.js]: Okay cool! Thanks for the update @mayank8318 :)

**Comment by mayank8318** [comment]
Ping @niranjan94 @CosmicCoder96 @ritikamotwani Please test and review :)

**Comment by mayank8318** [comment]
@geekyd Please review now :)

**Pull request merged**